pkg/rebuild/rebuild/rebuildremote.go

```diff
@@ -185,22 +185,14 @@ func RebuildRemote(ctx context.Context, input Input, id string, opts RemoteOptio
 			return errors.Wrap(err, "writing Dockerfile")
 		}
 	}
-	// NOTE: Ignore the local writer since GCS doesn't flush writes until Close.
-	// TODO: Could be resolved by adding ResourcePath() method.
-	_, imageUploadPath, err := opts.RemoteMetadataStore.Writer(ctx, Asset{Target: t, Type: ContainerImageAsset})
-	if err != nil {
-		return errors.Wrap(err, "creating dummy writer for container image")
-	}
-	_, rebuildUploadPath, err := opts.RemoteMetadataStore.Writer(ctx, Asset{Target: t, Type: RebuildAsset})
-	if err != nil {
-		return errors.Wrap(err, "creating dummy writer for rebuild")
-	}
+	imageUploadPath := opts.RemoteMetadataStore.URL(Asset{Target: t, Type: ContainerImageAsset}).String()
+	rebuildUploadPath := opts.RemoteMetadataStore.URL(Asset{Target: t, Type: RebuildAsset}).String()
 	build := makeBuild(t, dockerfile, imageUploadPath, rebuildUploadPath, opts)
```

Comment:
The LocatableAssetStore is, by design, implementation agnostic. However I think makeBuild is implicitly expecting these paths to be GCS paths.

Should we add a check here that the URL schema is GCS? I suspect that might break some tests that are using a filesystem assetstore for convenience, but it feels wrong to use the generic LocatableAssetStore in places where we actually want the specific GCSAssetStore

Reply:
So gsutil does, in fact, work with `file://` URLs but it's true that the interface is not necessarily enforcing the exact behavior we believe will make the system work correctly. Directionally, though, I think it does provide a clearer indication of where we're breaking out of the normal AssetStore usage pattern and allows us to think about encoding more in the type system in the future.

Reply:
Yeah this change is definitely an improvement, adding structure to the URLs allows the caller to at least inspect what kind of location is being returned.

The previous implementation was only providing an unstructured string which made it difficult for the caller to understand what they can do with the resource.